onnx/onnxifi_ext.h

```diff
@@ -0,0 +1,229 @@
+ * @retval ONNXIFI_STATUS_INVALID_POINTER The function call failed because
+ *                                        name or function argument is NULL.
+ * @retval ONNXIFI_STATUS_UNIDENTIFIED_NAME The function call failed because
+ *                                          the backend does not implement
+ *                                          the function identified by the name.
+ * @retval ONNXIFI_STATUS_INTERNAL_ERROR The function call failed because the
+ *                                       backend experienced an unrecovered
+ *                                       internal error.
+ * @retval ONNXIFI_STATUS_BACKEND_UNAVAILABLE The function call failed because
+ *                                            the backend was disconnected or
+ *                                            uninstalled from the system.
+ */
+
+ONNXIFI_PUBLIC ONNXIFI_CHECK_RESULT onnxStatus ONNXIFI_ABI
+onnxGetExtensionFunctionAddress(
+    onnxBackendID backendID,
+    const char* name,
+    onnxExtensionFunctionPointer* function);
+
+/* Extension function pointer declarations for dynamic loading */
+typedef ONNXIFI_CHECK_RESULT onnxStatus(
+    ONNXIFI_ABI* onnxSetIOAndRunGraphFunction)(
+    onnxGraph graph,
+    uint32_t inputsCount,
+    const onnxTensorDescriptorV1* inputDescriptors,
+    uint32_t outputsCount,
+    const onnxTensorDescriptorV1* outputDescriptors,
+    onnxMemoryFenceV1* outputFence);
+
+/**
+ * A combination of onnxSetIO and onnxRunGraph, functionally equals to first run
+ * onnxSetIO(graph, inputsCount, inputDescriptors, outputsCount,
+ * outputDescriptors), then run onnxRunGraph(graph, inputFence, outputFence)
+ *
+ * As two separate functions, it is difficult to do atomic evaluation.
+ * Therefore, we would like to unify this process and make it evaluable.
+ *
+ * @param graph - graph handle created by onnxInitGraph.
+ * @param inputsCount - number of elements in the inputDescriptors array.
+ * @param[in] inputDescriptors - descriptors of input tensors for the graph.
+ *                               Elements of this array must provide a location
+ *                               for each ValueInfoProto.name listed in
+ *                               ModelProto.graph.input of the ONNX graph.
+ *                               If inputsCount is non-zero, inputDescriptors
+ *                               pointer must be non-NULL.
+ * @param outputsCount - number of elements in the outputDescriptors array.
+ *                       Must be greater than zero.
+ * @param[in] outputDescriptors - descriptors of output tensors for the graph.
+ *                                outputDescriptors pointer must be non-NULL.
+ *                                Elements of this array must provide a location
+ *                                for each ValueInfoProto.name listed in
+ *                                ModelProto.graph.output of the ONNX graph.
+ * @param[in] inputFence - synchronization primitive that signals when graph
```

Comment:
do not forget to remove this from the comment, since it was removed from the function declaration.

Reply:
Yes!